spark/dl/src/test/scala/com/intel/analytics/bigdl/torch/AbsCriterionSpec.scala

```diff
@@ -20,14 +20,9 @@ import com.intel.analytics.bigdl.tensor.Tensor
 import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
 
 @com.intel.analytics.bigdl.tags.Serial
-class AbsCriterionSpec extends FlatSpec with BeforeAndAfter with Matchers{
-  before {
-    if (!TH.hasTorch()) {
-      cancel("Torch is not installed")
-    }
-  }
-
-  "A Abs Criterion " should "generate correct output and grad" in {
+class AbsCriterionSpec extends TorchSpec with BeforeAndAfter with Matchers{
+    "A Abs Criterion " should "generate correct output and grad" in {
+    torchCheck()
```

Comment:
I have added `BeforeAndAfter` and `Matchers` to the `TorchSpec`. Is it necessary to add `BeforeAndAfterAll` given we already have `BeforeAndAfter`?